examples/secret_storage/secret_storage.go

```diff
@@ -0,0 +1,572 @@
+	secret, err := sc.loadSecret(teamName, namespace)
+	if err != nil {
+		return "", err
+	}
+	mac := hmac.New(sha256.New, secret)
+	mac.Write([]byte(entryKey))
+	hmacEntryKey := mac.Sum(nil)
+	return hex.EncodeToString(hmacEntryKey), nil
+}
+
+func (sc *SecretKeyKVStoreAPI) PutEntry(teamName string, namespace string, entryKey string, entryValue string) (result keybase1.KVPutResult, err error) {
+	return sc.PutEntryWithRevision(teamName, namespace, entryKey, entryValue, 0)
+}
+
+func (sc *SecretKeyKVStoreAPI) PutEntryWithRevision(teamName string, namespace string, entryKey string, entryValue string, revision int) (result keybase1.KVPutResult, err error) {
+	var keyedValue map[string]string
+	if err = json.Unmarshal([]byte(entryValue), &keyedValue); err != nil {
+		return result, err
+	}
+
+	keyedValue[sc.keyKey] = entryKey
+	bytes, err := json.Marshal(keyedValue)
+	if err != nil {
+		return result, err
+	}
+
+	hmacEntryKey, err := sc.hmacKey(teamName, namespace, entryKey)
+	if err != nil {
+		return result, err
+	}
+
+	result, err = sc.api.PutEntryWithRevision(teamName, namespace, hmacEntryKey, string(bytes), revision)
+	if err != nil {
+		return result, err
+	}
+	result.EntryKey = entryKey
+	return result, err
+}
+
+func (sc *SecretKeyKVStoreAPI) DeleteEntry(teamName string, namespace string, entryKey string) (result keybase1.KVDeleteEntryResult, err error) {
+	return sc.DeleteEntryWithRevision(teamName, namespace, entryKey, 0)
+}
+
+func (sc *SecretKeyKVStoreAPI) DeleteEntryWithRevision(teamName string, namespace string, entryKey string, revision int) (result keybase1.KVDeleteEntryResult, err error) {
+	hmacEntryKey, err := sc.hmacKey(teamName, namespace, entryKey)
+	if err != nil {
+		return result, err
+	}
+	result, err = sc.api.DeleteEntryWithRevision(teamName, namespace, hmacEntryKey, revision)
+	if err != nil {
+		return result, err
+	}
+	result.EntryKey = entryKey
+	return result, err
+}
+
+func (sc *SecretKeyKVStoreAPI) GetEntry(teamName string, namespace string, entryKey string) (result keybase1.KVGetResult, err error) {
+	hmacEntryKey, err := sc.hmacKey(teamName, namespace, entryKey)
+	if err != nil {
+		return result, err
+	}
+	result, err = sc.api.GetEntry(teamName, namespace, hmacEntryKey)
+	if err != nil {
+		return result, err
+	}
+	result.EntryKey = entryKey
+	return result, err
+}
+
+func (sc *SecretKeyKVStoreAPI) ListNamespaces(teamName string) (keybase1.KVListNamespaceResult, error) {
+	return sc.ListNamespaces(teamName)
+}
+
+func (sc *SecretKeyKVStoreAPI) ListEntryKeys(teamName string, namespace string) (result keybase1.KVListEntryResult, err error) {
+	keys, err := sc.api.ListEntryKeys(teamName, namespace)
+	if err != nil {
+		return result, err
+	}
+	tmp := keys.EntryKeys[:0]
+	for _, e := range keys.EntryKeys {
+		if strings.HasPrefix(e.EntryKey, "_") {
+			continue
+		}
+
+		get, err := sc.api.GetEntry(teamName, namespace, e.EntryKey)
```

Comment:
rather than fetch each and every entry key individually, would it be easier to store all of our metadata in a single entryKey in the namespace? something like:
```
entryKey = _metadata
entryValue = `{ secret: 'asefsafseafasef', keys: ['laser', 'somethingelse', ...]}`
```
and then listing would be as simple as fetching this entry and iterating through the list. bonus: it also gets rid of `keyKey`.

Reply:
slight risk of things falling out of sync, but that doesn't seem like a big deal for an example. especially if we make a note of it somewhere.

Reply:
that's definitely another way to deal with ListEntryKeys. i'm inclined to keep the approach the same as in the python example. we also chose this way because it is easier to keep in sync. someone who actually is deploying this bot can decide what their performance/consistency requirements are.